### PR TITLE
Further improve modularization

### PR DESCRIPTION
**Pull request by eric-hansen**
This doesn't have configuring things yet, but wanted to open this for discussion.

### CODE REVIEW

File: lua/which-key-config/init.lua [hunk 1]
```diff
@@ -1,6 +1,6 @@
-local wk = Vapour.utils.plugins.exists('which-key.nvim')
+Vapour.utils.plugins.packadd('which-key.nvim')
```
Comment (eric-hansen) [inline lua/which-key-config/init.lua:1]: It might be beneficial to do this for anything that's lazyloaded, but not sure.

**Comment by hackorum** [comment]
Does the autocomplete work for you now?

**Comment by eric-hansen** [comment]
> Does the autocomplete work for you now?

Nope but this does resolve some other issues.

I'm thinking lazy-loading isn't the best approach honestly.  But that's my 2 cents.

I've worked on migrating a fresh nvim set up using Paq instead of Packer which doesn't support lazy loading.  Reason being if the plugin "needs" it then what is the purpose of the plugin basically.  After some tweaking around I've gotten nvim-compe to work as expected there, so I'll look and compare what's in Vapour with what I got otherwise, but I feel lazy-loading plays the biggest role in this.

**Comment by eric-hansen** [comment]
So I was able to get completion to work now.  The main issue being it seems the way `lua/compe-config/init.lua` was set up.  I'll re-add the lazy loading to see if that makes a difference and if it still works I'll leave that in again.

**Comment by eric-hansen** [comment]
Alright so this is good now and resolves #5 and #10 

Most plugins are disabled by default as it should be up to the user imo to decide if a plugin is useful for them or not.  But if you disagree let me know and I'll make appropriate changes.

**Pull request merged**